### DISCUSSION TITLE
Virtual Chassis Interface naming, what is recommended to use?

### DISCUSSION THREAD

**Discussion by lonelyadmin** [netbox-community/netbox#5794] [Q&A] [answered]
I'm adding Cisco switches that are in a VSS pair. Before configuring VSS, each switch has Te1/1-48. After configuring VSS the interfaces on the secondary switch become Te2/1-48. When I add the second switch to the virtual chassis in netbox, should I use Te1 or Te2 for defining interfaces?

Or should I just scrap the config name of the interface and use 1-48? What's the general rule on this sort of thing?

**Comment by jhammond-git** [comment]
My general rule is to name them how they're named when I'm configuring them.  If you do a show run and they come up as Te2/1-48, then name them that on the device.

**Reply by lonelyadmin** [reply]
Thanks, that's what I ended up doing. It's how I reference them in every other system so it's probably better anyway.

**Comment by candlerb** [comment, marked as answer]
Before you create the virtual chassis, on the second device rename `Te1/[1-48]` to `Te2/[1-48]`.  There's a bulk interface rename function for that (*).  Then combine them.

If you don't do this, then you'll end up with two interfaces with the same name (but different IDs in netbox and relating to different physical ports), which is very confusing.

(*) assuming that the interfaces already exist, because you put them in the Device Type template. If you didn't do this, then just create interfaces `Te2/[1-48]` using interface create with a range pattern.

**Reply by lonelyadmin** [reply]
That's exactly how I ended up doing it. I wasn't sure at first if configuring a virtual chassis would do that automatically...it doesn't.

**Reply by jhammond-git** [reply]
Sadly, no.  Since every platform does their numbering differently, it'd be a hard ask to have Netbox keep up with vendor bits like that.

**Reply by ljb2of3** [reply]
It would be interesting if netbox supported something like jinja variables in interface names. On my aruba switches interfaces are 1/1/[1-48] for member 1, 2/1/[1-48] for member 2, and so on... so if I could define something like {{ member_id }}/1/[1-48] that would be cool. The idea being that for an unstacked switch it would just default to 1, but if I create a virtual chassis then the interface names get automatically updated.

Maybe more trouble than it's worth, but an idea nonetheless.

I've resorted to creating my device templates with interfaces like X/1/[1-48] then doing bulk renames as I create each switch, but it would be nice to skip that step.
